examples/zookeeper/template.json

```diff
@@ -0,0 +1,265 @@
+    {
+      "kind": "Service",
+      "apiVersion": "v1beta3",
+      "metadata": {
+        "name": "zookeeper-2",
+        "labels": {
+          "name": "zookeeper-2"
+        }
+      },
+      "spec": {
+        "ports": [
+          {
+            "name": "client",
+            "port": 2181,
+            "targetPort": 2181
+          },
+          {
+            "name": "followers",
+            "port": 2888,
+            "targetPort": 2888
+          },
+          {
+            "name": "election",
+            "port": 3888,
+            "targetPort": 3888
+          }
+        ],
+        "selector": {
+          "server-id": "2"
+        }
+      }
+    },
+    {
+      "kind": "Service",
+      "apiVersion": "v1beta3",
+      "metadata": {
+        "name": "zookeeper-3",
+        "labels": {
+          "name": "zookeeper-3"
+        }
+      },
+      "spec": {
+        "ports": [
+          {
+            "name": "client",
+            "port": 2181,
+            "targetPort": 2181
+          },
+          {
+            "name": "followers",
+            "port": 2888,
+            "targetPort": 2888
+          },
+          {
+            "name": "election",
+            "port": 3888,
+            "targetPort": 3888
+          }
+        ],
+        "selector": {
+          "server-id": "3"
+        }
+      }
+    },
+    {
+      "kind": "Pod",
+      "id": "zookeeper-1",
+      "apiVersion": "v1beta1",
+      "labels": {
+        "name": "zookeeper",
+        "server-id": "1"
+      },
+      "desiredState": {
+        "manifest": {
+          "id": "zookeeper-1",
+          "containers": [
+            {
+              "name": "server",
+              "image": "${ZOOKEEPER_IMAGE}",
+              "env":[
+                { "name": "SERVER_ID", "value": "1" }
+              ],
+              "ports":[
+                {
+                  "containerPort": 2181
+                },
+                {
+                  "containerPort": 2888
+                },
+                {
+                  "containerPort": 3888
+                }
+              ]
+            }
+          ]
+        }
+      }
+    },
+    {
+      "kind": "Pod",
+      "id": "zookeeper-2",
+      "apiVersion": "v1beta1",
+      "labels": {
+        "name": "zookeeper",
+        "server-id": "2"
+      },
+      "desiredState": {
+        "manifest": {
+          "id": "zookeeper-2",
+          "containers": [
+            {
+              "name": "server",
+              "image": "${ZOOKEEPER_IMAGE}",
+              "env":[
+                { "name": "SERVER_ID", "value": "2" }
+              ],
+              "ports":[
+                {
+                  "containerPort": 2181
+                },
+                {
+                  "containerPort": 2888
+                },
+                {
+                  "containerPort": 3888
+                }
+              ]
+            }
+          ]
+        }
+      }
+    },
+    {
+      "kind": "Pod",
```

Comment:
Cluster membership and identity - replacement of a controller that didn't think it was 1 or 2 or 3 caused the cluster to die because they can't reach quorum (since zookeeper pre 3.5 doesn't handle membership change gracefully, and 3.5 is still not ready)

----- Original Message -----
> > +                  "containerPort": 2181
> > +                },
> > +                {
> > +                  "containerPort": 2888
> > +                },
> > +                {
> > +                  "containerPort": 3888
> > +                }
> > +              ]
> > +            }
> > +          ]
> > +        }
> > +      }
> > +    },
> > +    {
> > +      "kind": "Pod",
> 
> Why wasn't this a replication controller?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/1965/files#r29721644